gpii/node_modules/preferencesServer/test/prefsSafesTests.js

```diff
@@ -0,0 +1,715 @@
+        expect: 2,
+        components: {
+            getRequest: {
+                type: "kettle.test.request.http",
+                options: {
+                    path: "/prefssafe",
+                    method: "POST",
+                    port: 8081
+                }
+            }
+        },
+        sequence: [{
+            func: "{getRequest}.send",
+            args: fixture.postBody
+        }, {
+            event: "{getRequest}.events.onComplete",
+            listener: "gpii.tests.preferencesServer.testResponseLeftHand",
+            args: ["{arguments}.0", fixture.expected, "{getRequest}.nativeResponse.statusCode", expectedStatusCode]
+        }]
+    };
+};
+
+gpii.tests.preferencesServer.prefsSafes.postSafe.successFixtures = [
+    {
+        name: "POST: Create a new preferences safe.",
+        url: "%prefsSafeId",
+        postBody: {
+            "type": "prefsSafe",
+            "schemaVersion": "0.1",
+            "prefsSafeType": "user",
+            "name": "Prefs Three Thousand",
+            "email": null,
+            "preferences": {
+                "flat": {
+                    "name": "bit of stuff",
+                    "contexts": {
+                        "gpii-default": {
+                            "name": "Default preferences",
+                            "preferences": {
+                                "http://registry.gpii.net/common/onScreenKeyboard/enabled": false
+                            },
+                            "metadata": []
+                        }
+                    },
+                    "metadata": []
+                }
+            }
+        },
+        expected: {
+            "type": "prefsSafe",
+            "schemaVersion": "0.1",
+            "prefsSafeType": "user",
+            "name": "Prefs Three Thousand",
+            "email": null,
+            "preferences": {
+                "flat": {
+                    "name": "bit of stuff",
+                    "contexts": {
+                        "gpii-default": {
+                            "name": "Default preferences",
+                            "preferences": {
+                                "http://registry.gpii.net/common/onScreenKeyboard/enabled": false
+                            },
+                            "metadata": []
+                        }
+                    },
+                    "metadata": []
+                }
+            }
+        }
+    }
+];
+
+/////////////////////// TESTING prefsSafes List (GET) ////////////////////////////
+fluid.registerNamespace("gpii.tests.preferencesServer.prefsSafes.list");
+
+gpii.tests.preferencesServer.prefsSafes.list.buildTestDef = function (fixture) {
+    var expectedStatusCode = fixture.expectedStatusCode || 200;
+    return {
+        name: fixture.name,
+        expect: 2,
+        components: {
+            getRequest: {
+                type: "kettle.test.request.http",
+                options: {
+                    path: "/prefssafes",
```

Comment:
To re-produce the error I ran into:

1. Open a terminal, in the /universal root directory, run `NODE_ENV=gpii.config.preferencesServer.standalone.development.manualTesting.json5 node gpii.js` to start the prefs server;
2. Use curl command or other tools to send a GET request to http://localhost:8081/prefssafes;
3. The response received is:
```
{
    "isError": true,
    "message": "Cannot read property 'rows' of undefined"
}
```
Meanwhile, GPII started in the terminal terminates with this error:
```
FATAL ERROR: Uncaught exception: Cannot read property 'rows' of undefined
TypeError: Cannot read property 'rows' of undefined
    at gpii.dbOperation.dbDataStore.cleanUpDocList (/Users/gpii/node_modules/universal/gpii/node_modules/gpii-db-operation/src/DbDataStoreUtils.js:159:14)
```

Reply:
This seems to be working for me now. I also tested going in to fauxton and deleting all the prefssafes to test with zero rows.  Your couch views might need to get to recreated?